examples/statistics/time_series_histogram.py

```diff
@@ -80,13 +80,15 @@
 cmap = copy(plt.cm.plasma)
 cmap.set_bad(cmap(0))
 h, xedges, yedges = np.histogram2d(x_fine, y_fine, bins=[400, 100])
-axes[1].pcolormesh(xedges, yedges, h.T, cmap=cmap, norm=LogNorm(vmax=1.5e2))
+axes[1].pcolormesh(xedges, yedges, h.T, cmap=cmap,
+                   norm=LogNorm(vmax=1.5e2), rasterized=True)
 axes[1].set_title("2d histogram and log color scale")
 
 # Same data but on linear color scale
-axes[2].pcolormesh(xedges, yedges, h.T, cmap=cmap, vmax=1.5e2)
+axes[2].pcolormesh(xedges, yedges, h.T, cmap=cmap, vmax=1.5e2, rasterized=True)
 axes[2].set_title("2d histogram and linear color scale")
 toc = time.time()
-print(f"{toc-tic:.2f} sec. elapsed")
+print(f"{toc-tic:.3f} sec. elapsed")
 
+fig.tight_layout()
```

Comment:
Constrained layout?  I guess the other nit, is that the poolers *should* have colorbars...

Reply:
I figured I'd use `tight_layout()` to reduce the whitespace in the margins. Is this a bad practice?

Reply:
Well if you add the colorbars, you will find that constrained_layout works better...

Reply:
Ahh, I didn't even realize `constrained_layout` was even a thing. Only ever used `tight_layout` before.

Reply:
Well I wrote it, so I try to twist peoples' arms to use it ;-)

Reply:
Very cool! Does it actually use CSP or LP to optimize the layout, or is it like a heuristic solver?

Reply:
Not sure what CSP or LP are :wink:.  It uses a linear constraint solver (kiwi solver) and relatively straight forward constraints....

Reply:
Ahh; CSP = Constraint Satisfaction Problem/Programming, and LP = Linear Programming. They're very broad classes of numerical techniques to optimize some objective under constraints. Looks like kiwi solver is based off [an LP algorithm](https://constraints.cs.washington.edu/solvers/cassowary-tochi.pdf), so I guess you know more about it than you think :smile: